src/gpodder/extensions.py

```diff
@@ -378,18 +379,41 @@ def _find_extensions(self):
             builtins = os.path.join(gpodder.prefix, 'share', 'gpodder',
                 'extensions', '*.py')
             user_extensions = os.path.join(gpodder.home, 'Extensions', '*.py')
-            self.filenames = glob.glob(builtins) + glob.glob(user_extensions)
 
-        # Let user extensions override built-in extensions of the same name
+            # sort filenames so that if duplicates are found in the same folder,
+            # the highest priority (lowest number) will always be used.
+            self.filenames = sorted(glob.glob(builtins), reverse=True) \
+                + sorted(glob.glob(user_extensions), reverse=True)
+
+        # Let user extensions override built-in extensions of the same name.
+        # This inherently happens because we search the user extensions folder second,
+        # and the entries are put in the extensions dict by their name field.
         for filename in self.filenames:
             if not filename or not os.path.exists(filename):
                 logger.info('Skipping non-existing file: %s', filename)
                 continue
 
             name, _ = os.path.splitext(os.path.basename(filename))
-            extensions[name] = filename
 
-        return sorted(extensions.items())
+            # extensions with no priority get lowest priority, 99
+            priority = 99
+            m = re.fullmatch(r'^([0-9]*)_(.+)', name)
+            if m:
+                if m.group(1):
+                    # get ordering prefix
+                    priority = int(m.group(1))
+                # strip ordering prefix from name (or leading _)
+                name = m.group(2)
+            if name in extensions:
+                if filename.startswith(gpodder.home) == extensions[name][1].startswith(gpodder.home):
+                    logger.warning("extension at %s will be ignored in favor of %s", extensions[name][1], filename)
+                else:
+                    logger.info("extension at %s will be ignored in favor of %s", extensions[name][1], filename)
+            extensions[name] = (priority, filename)
+
+        # sort by priority - extensions with same priority will retain the order
+        # in which they were found.
+        return sorted(extensions.items(), key=lambda i: (i[1][0], i[0]))
```

Comment:
The comment is actually wrong:

> extensions with same priority will retain the order in which they were found

In fact, we use [`glob.glob()`](https://docs.python.org/3/library/glob.html#glob.glob) when finding the extensions, and the Python docs there say:

> [...] although results are returned in arbitrary order

So it's a good thing they don't retain the order in which they were found. The code did the right thing and sort by name for same-priority items, but the comment was misleading and the code was hard to read (IMHO).

The `key=` function sorts by priority and then by name. This could be made be clearer like this:

```suggestion
        # sort by priority - extensions with same priority will be sorted by name
        def sort_key(kv):
            name, (priority, filename) = kv
            return (priority, name)
            
        return sorted(extensions.items(), key=sort_key)
``````

Reply:
The comment used to be correct, I just neglected to update it. I will update it.

Suggestion looks fine by me, whatever works. Seems to work fine from quick testing.